### ISSUE TITLE
problem with reference date in scheduledDays computation

### ISSUE THREAD

**Issue by bergmannjg** [public-transport/hafas-client#277]
Given the following TripSearch response

```
{
    ... 
    "svcResL": [
        {
            "meth": "TripSearch",
            "res": {
                ...
                "outConL": [
                    {
                        "date": "20221115",
                        "sDays": {
                            "sDaysR": "runs 15. Nov until 10. Dec 2022 ",
                            "sDaysB": "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF9FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF807FFFFFF0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
                        },
                    ...
                ],
                "fpB": "20211212",
                "fpE": "20231209",
        ...
        }
    ]
}
```

hafas-client computes the last scheduledDay  as _2022-12-30_, the correct value should be _2022-12-10_ as in field _sDaysR_.

hafas-client uses the year of _"date": "20221115"_ as the reference date.

https://github.com/public-transport/hafas-client/blob/61fc2293fb41df5a7c43a202df8707134cd51759/parse/journey.js#L60-L63

https://github.com/public-transport/hafas-client/blob/61fc2293fb41df5a7c43a202df8707134cd51759/parse/journey.js#L10-L15

I would suggest to take the reference date from _"fpB": "20211212"_ as year/month/day.

**Comment by derhuerst** [comment]
> hafas-client computes the last scheduledDay as `2022-12-30`, the correct value should be `2022-12-10` as in field `sDaysR`.

Good catch!

> I would suggest to take the reference date from `"fpB": "20211212"` as year/month/day.

This seems to be the intended way, yes.

**Comment by derhuerst** [comment]
Just found this weird example, can you make sense of it?

```json5
{
	// […]
	"fpB": "20211212",
	"fpE": "20231209",
	// […]
	"sDays": {
	 	"sDaysR": "runs 16. Nov 2022 ",
	 	"sDaysB": "000000000000000000000000000000000000000000000000000000000000000000000000000000000001F0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
	},
}
```

**Comment by bergmannjg** [comment]
sDaysB  decodes the range from 2022-11-12 to 2022-11-16

if current day is in the sDaysB  range then sDaysR range starts from current day '2022-11-16'

**Comment by derhuerst** [comment]
> if current day is in the sDaysB range then sDaysR range starts from current day '2022-11-16' [but `sDaysB` *does not*]

That is a weird API!

**Comment by derhuerst** [comment]
Fix published as `hafas-client@5.26.2`. 🎉
